gdal/src/it/scala/geotrellis/contrib/vlm/SubsceneReadingIT.scala

```diff
@@ -27,8 +27,27 @@ import org.scalatest.{FunSpec, Matchers}
 class SubsceneReadingIT extends FunSpec with Matchers with LazyLogging {
   import Timing._
 
-  val sample = "http://s3-us-west-2.amazonaws.com/radiant-nasa-iserv/2014/02/14/IP0201402141023382027S03100E/IP0201402141023382027S03100E-COG.tif"
-  // val sample = "s3://radiant-nasa-iserv/2014/02/14/IP0201402141023382027S03100E/IP0201402141023382027S03100E-COG.tif"
+  /*
+   * APOLOGY
+   *
+   * In order to find metadata-containing sidecar files, GDAL's
+   * GeoTiff driver takes a listing of the directory containing the
+   * file in question.  This succeeds with `/vsis3/` URIs because S3
+   * supports that kind of thing (the first file in the listing is the
+   * GeoTiff itself, so GDAL extracts the metadata from that file).
+   * For `/vsicurl/` URIs, GDAL hits the URI of the directory and
+   * hopes to get back a sensible response, but S3's http server does
+   * not give one.  It then begins generating filenames, and attempts
+   * to `stat` them which results in a delay.
+   *
+   * For that reason, an http link cannot be used to (fairly) test
+   * GDAL's performance.
+   *
+   * However, one cannot use an S3 link to test GeoTrellis'
+   * performance because of
+   * https://github.com/locationtech/geotrellis/issues/2889
+   */
+  val sample = "https://s3-us-west-2.amazonaws.com/radiant-nasa-iserv/2014/02/14/IP0201402141023382027S03100E/IP0201402141023382027S03100E-COG.tif"
```

Comment:
@jamesmcclain Wow.... super helpful info!

Reply:
I am hoping to have the time/energy this evening to submit a pull request to GDAL to address this issue.

Reply:
Super nuanced semantics I don't think I'd ever have figured out.

Reply:
I used `gdb` :smile: